core/store/src/adapter/trie_store.rs

```diff
@@ -173,6 +173,17 @@ impl<'a> TrieStoreUpdateAdapter<'a> {
     }
 }
 
+/// Get the `ShardUId` mapping for child_shard_uid. If the mapping does not exist, map the shard to itself.
+/// Used by Resharding V3 for State mapping.
+pub fn get_shard_uid_mapping(store: &Store, child_shard_uid: ShardUId) -> ShardUId {
```

Comment:
Could we push this pub function to `TrieStoreAdapter`? In cold_storage.rs we can call `hot_store.trie_store().get_shard_uid_mapping(...)`

Reply:
I would like to but it is problematic, please see https://github.com/near/nearcore/pull/12232#discussion_r1804810508.

Reply:
Aah okay! Yes, we don't want to be cloning store I guess. Could you add a quick comment just so that next time we have context for why it's where it is?

Reply:
Added here: f46714f3c903c73ed30610fa351bc74a06a96f0a